src/helpers/reciever.js

```diff
@@ -0,0 +1,330 @@
+      // ask dynamo what is the lastest hash it received
+      let cacheLatestItem = this.cacheGetLatestItem(); //[HASH].S //rename to lastest hash
+      let cacheLatestHash = "";
+      if (cacheLatestItem === null) {
+        // no cache found, pull every thing and set
+        this.CWAData = await this.getAllItems();
+        this.latestItem = dynamoLatestItem;
+        this.cacheSaveData();
+        return true;
+      } else {
+        cacheLatestHash = cacheLatestItem[HASH];
+        this.CWAData = this.cacheGetAllData();
+      }
+
+      if (DynamoHash === cacheLatestHash) {
+        return this.updateReleases() // already synced
+      } else if (
+        parseInt(dynamoLatestItem[COMMIT_DATE]) >=
+        parseInt(cacheLatestItem[COMMIT_DATE])
+      ) {
+        /// if hashes dont match call getBatchItem with local hash and update local hash with new data
+        var newItems = await this.getBatchItem(
+          cacheLatestItem[COMMIT_DATE],
+          dynamoLatestItem[COMMIT_DATE]
+        );
+        Object.keys(newItems).forEach((key) => {
+          if (this.CWAData[key] === undefined) {
+            // new testCase
+            this.CWAData[key] = {};
+          }
+          //already have this testCase
+          Object.keys(this.CWAData[key]).forEach((metric) => {
+            if (this.CWAData[key][metric] === undefined) {
+              // new metric
+              this.CWAData[key][metric] = [];
+            }
+            //already have this metric
+            this.CWAData[key][metric].push(...newItems[key][metric]);
+          });
+        });
+        this.latestItem = dynamoLatestItem;
+        this.cacheSaveData();
+        return this.updateReleases() // now synced
+      }
+      // website is ahead of dynamo
+      //clear cache and retry 
+      this.cacheClear()
+      this.update()
+    } catch (err) {
+      console.log(`ERROR:${err}`);
+      alert(`ERROR:${err}`);
+      if (this.cacheGetLatestItem === undefined) {
+        return false
+      }
+      this.CWAData = this.cacheGetAllData()
+      return false //couldnt sync
+    }
+  }
+  async updateReleases(){
+    //release backtracking
+    try{
+      var allReleases = await this.getAllReleases()
+      allReleases.forEach((item)=>{
+        if (this.ReleaseMap[item[HASH].S]){
+          //old
+          return
+        }
+        //new 
+        Object.keys(item[RESULTS].M).forEach((testCase,testCaseValue)=>{
+          Object.keys(testCaseValue).forEach((metric)=>{
+            //update link and release tag
+            this.CWAData[testCase][metric][IS_RELEASE] = true
+            // this.CWAData[testCase][metric][LINK] = ""
+            // this.CWAData[testCase][metric][HASH] = 
+            console.log("Updated",item[HASH].S)
+          })
+          
+        })
+        this.ReleaseMap[item[HASH].S] = true
+        this.cacheSaveData()
+      })
+      return true
+    }
+    catch(err){
+      alert(`ERROR:${err}`);
```

Comment:
I don't think we should be doing this.

Reply:
By this, you mean the alert? I agree, I will switch it with a console.log if that is sufficient